crates/network-scanner/src/mdns.rs

```diff
@@ -0,0 +1,86 @@
+use mdns_sd::ServiceEvent;
+
+use crate::{
+    task_utils::{PortReceiver, TaskManager},
+    ScannerError,
+};
+
+const META_QUERY: &str = "_services._dns-sd._udp.local.";
+
+pub fn mdns_query_scan(
+    service_deamon: mdns_sd::ServiceDaemon,
+    task_manager: TaskManager,
+    entire_duration: std::time::Duration,
+    single_query_duration: std::time::Duration,
+) -> Result<PortReceiver, ScannerError> {
+    let receiver = service_deamon.browse(META_QUERY)?;
+    let service_deamon_clone = service_deamon.clone();
+
+    let (result_sender, result_receiver) = tokio::sync::mpsc::channel(255);
+
+    task_manager
+        .with_timeout(entire_duration)
+        .when_finished(move || {
+            tracing::debug!("mdns meta query finished");
+            while let Err(e) = service_deamon_clone.stop_browse(META_QUERY) {
```

Comment:
using retry as indicated in it's doc
```
/// Stops searching for a specific service type.
    ///
    /// When an error is returned, the caller should retry only when
    /// the error is `Error::Again`, otherwise should log and move on.
    pub fn stop_browse(&self, ty_domain: &str) -> Result<()> {
        self.send_cmd(Command::StopBrowse(ty_domain.to_string()))
    }
```